src/Console/Command/ObfuscateData.php

```diff
@@ -0,0 +1,305 @@
+    {
+        $this->info("Overwriting post author data...");
+        $faker = Faker\Factory::create();
+        $results = DB::select('posts.*')
+            ->from('posts')
+            ->execute($this->db)
+            ->as_array();
+        $overwrittenCount = 0;
+        $resultsCount = count($results);
+        foreach ($results as $row) {
+            $postEntity = $this->postRepository->getEntity($row);
+            $fakeEmail = $faker->safeEmail;
+            $fakeName = $faker->firstName." ".$faker->lastName;
+            $postEntity->setState([
+                'author_realname'=> $fakeName,
+                'author_email'=> $fakeEmail,
+            ]);
+            $overwrittenCount += $this->postRepository->update($postEntity);
+        }
+        if ($overwrittenCount == $resultsCount) {
+            $this->info("updated ".$overwrittenCount." records.\n");
+            return $overwrittenCount;
+        } else {
+            $this->info("failed to overwrite all records.\n");
+            return false;
+        }
+    }
+
+    private function overwriteSiteConfig()
+    {
+        $this->info("\nRemoving general settings data...");
+        $siteConfig = $this->configRepository->get('site');
+
+        //prepend with OBFUSCATED if not already prepended
+        $newSiteName = $siteConfig->name;
+        if (substr($siteConfig->name, 0, 11) !== 'OBFUSCATED:') {
+            $newSiteName = 'OBFUSCATED: '.$newSiteName;
+        }
+        $newSiteDescription = $siteConfig->description;
+        if (substr($newSiteDescription, 0, 11) !== 'OBFUSCATED:') {
+            $newSiteDescription = 'OBFUSCATED: '.$newSiteDescription;
+        }
+
+        $siteConfig->setState([
+            'name' => $newSiteName,
+            'description' => $newSiteDescription,
+            'email' => 'admin@ushahidi.com',
+            //TODO: reset these?
+            //'cient_url' => '??',
+            // 'url' => '??'
+        ]);
+        $this->configRepository->update($siteConfig);
+        $this->info("\nDone.");
+    }
+
+    private function removeDataProviderValues()
+    {
+        $this->info("\nORemoving dataProvider data...");
+        $dataProviderConfig = $this->configRepository->get('data-provider');
+        //TODO: walk through each key instead?
+        //TODO:  otherwise set them to defaults?
+        $dataProviderConfig->setState([
+            'url' => '',
+            'providers' => [
+                'smssync' => false,
+                'email' => false,
+                'twilio' => false,
+                'nexmo' => false,
+                'twitter' => false,
+                'frontlinesms' => false
+            ],
+            'frontlinesms'=> [],
+            'nexmo'=> [],
+            'smssync'=> [],
+            'twilio'=> [],
+            'email'=> [],
+        ]);
+        $this->configRepository->update($dataProviderConfig);
+        $this->info("\nSet all dataproviders to false.");
```

Comment:
I might be wrong but I think artisan add newlines for you so you can probably drop all the `\n`s